### PR TITLE
CI: add a workflow to publish on PyPI

### PR DESCRIPTION
**Pull request by mrakitin**
Based on https://github.com/bluesky/databroker/blob/master/.github/workflows/publish-pypi.yml.

### CODE REVIEW

File: .github/workflows/publish-pypi.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# This workflows will upload a Python Package using flit when a release is created
```
Comment (jklynch) [inline .github/workflows/publish-pypi.yml:1]: `workflow`

File: .github/workflows/publish-pypi.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install wheel twine setuptools
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: __token__
+        # The PYPI_PASSWORD must be a pypi token with the "pypi-" prefix with sufficient permissions to upload this package
+        # https://pypi.org/help/#apitoken
+        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
```
Comment (jklynch) [inline .github/workflows/publish-pypi.yml:30]: the nslsii repository has a `secrets.PYPI_API_TOKEN`

Comment (mrakitin) [reply .github/workflows/publish-pypi.yml:30]: Is it set per repo or per org?

Comment (mrakitin) [reply .github/workflows/publish-pypi.yml]: Changed.

**Comment by mrakitin** [comment]
Thanks, @jklynch!

**Pull request merged**